lib/solidus_subscriptions/checkout.rb

```diff
@@ -32,14 +32,8 @@ def process
     private
 
     def create_order
-      ::Spree::Order.create(
-        user: installment.subscription.user,
-        email: installment.subscription.user.email,
-        store: installment.subscription.store || ::Spree::Store.default,
-        subscription_order: true,
-        subscription: installment.subscription,
-        currency: installment.subscription.currency
-      )
+      extra_attributes = SolidusSubscriptions.configuration.order_creation_extra_attributes
+      SolidusSubscriptions.configuration.order_creation_class.new(installment, extra_attributes).call
```

Comment:
This might be a matter of personal preference, but I'm not too fond of having a configurable class (`order_creation_class`) and then an additional configuration option (`order_creation_extra_attributes`) that the configurable class is supposed to use.

I think it's potentially confusing, because a custom order creation class may very well disregard the extra attributes completely, but the user will be under the impression that the attributes are being set because they've been configured in a separate option.

The way we've solved this in the past is by having the extra attributes as a method on the order creation class, so that the user can create a custom order creator that simply inherits from the original one and adds some extra attributes.

Another approach would be to add a configuration option to the "standard" order creation class, although I'd rather keep the configuration centralized in one place and just have users write their own class.

Let me know your thoughts! 🙂

Reply:
That makes a lot of sense. So I will be following the approach of adding a `extra_attributes` method to the `OrderCreator` itself